contracts/abstracted_account.algo.ts

```diff
@@ -84,25 +131,31 @@ export class AbstractedAccount extends Contract {
    * @param controlledAddress The address of the abstracted account. If zeroAddress, then the address of the contract account will be used
    * @param admin The admin for this app
    */
-  createApplication(controlledAddress: Address, admin: Address): void {
-    verifyAppCallTxn(this.txn, {
-      sender: { includedIn: [controlledAddress, admin] },
-    });
-
+  @abimethod({ onCreate: 'require'})
+  createApplication(controlledAddress: arc4.Address, admin: arc4.Address) {
+    // verifyAppCallTxn(this.txn, {
```

Comment:
Just an FYI, Puya has `assertMatch` which will give the same functionality as `verify...Txn`